### PR TITLE
OrbitControls: Support frame rate independent `autoRotate`.

### LINKED ISSUE #26471: OrbitControls are framerate dependent
### Description

The OrbitControl provided with three.js is framerate dependent.
Running at 30fps will spin half as fast at 60fps for example. 
This can be accounted for by taking into account the time between frames.

### Reproduction steps

1. Make a default scene with OrbitControls and autoRotate.
2. Limit the FPS to different amounts and observe the rotation effect.

### Code

```js
// code goes here
```

### Live example

[jsfiddle](https://jsfiddle.net/issung/wzjqn9r7/16/)

### Screenshots

_No response_

### Version

latest

### Device

Desktop, Mobile, Headset

### Browser

Chrome, Firefox, Safari, Edge

### OS

Windows, MacOS, Linux, Android, iOS

### PR DESCRIPTION
**Pull request by Issung**
Fixed #26471.

Updates the OrbitControls `getAutoRotationAngle()` function to take into account clock delta so that `autoRotate` can become framerate independent.

### CODE REVIEW

**Comment by Issung** [comment]
See my first commit on the branch to see some of my logic expressed in some comments, I'm pretty sure the math is equivalent.

**Comment by Issung** [comment]
Some unexpected failures here, is this typical?

**Comment by WestLangley** [comment]
You can do this in your app. See https://github.com/mrdoob/three.js/issues/24956#issuecomment-1314536979.

To maintain the current default rotation rate, you would do this:
```
controls.autoRotateSpeed = 120 * clock.getDelta();
```

**Comment by Issung** [comment]
I think it would be good to make it FPS independent by default for all users of the class :)

**Comment by Mugen87** [comment]
Another discussion about this topic:

https://discourse.threejs.org/t/speed-of-orbitcontrols-autorotate-not-related-to-fps/43910

I guess I would prefer a solution within `OrbitControls` to avoid further confusion. IMO, a FPS independent solution is the more appropriate default behavior.

File: examples/jsm/controls/OrbitControls.js [hunk 1]
```diff
@@ -470,9 +472,7 @@ class OrbitControls extends EventDispatcher {
 		const pointerPositions = {};
 
 		function getAutoRotationAngle() {
-
-			return 2 * Math.PI / 60 / 60 * scope.autoRotateSpeed;
-
+			return (2 * Math.PI / 60 * scope.autoRotateSpeed) * _clock.getDelta();
```
Comment (Mugen87) [inline examples/jsm/controls/OrbitControls.js:475]: The problem with using `THREE.Clock` is that when users switch to another tab and then come back to the application after a while, you end up with a huge time delta value that leads to a visible jump. Compare the PRs version with prod to see this effect:

https://raw.githack.com/Issung/three.js/%2326471-orbitcontrols-framerate-independent/examples/webgl2_volume_instancing.html
https://threejs.org/examples/webgl2_volume_instancing

The controls now also produce large time delta values when users enable - disable - enable auto rotation.

**Comment by Mugen87** [comment]
> Some unexpected failures here, is this typical?

This PR changes the behavior of `autoRotate` which means the screenshots are rendered differently (and thus the E2E test fails). The time delta value is `0` in the first frame which means `getAutoRotationAngle()` returns `0` as well.

A behavior change is not a problem in principle but a) we have to know them and b) inform the user via the migration guide is we decide to make this change.

**Comment by Issung** [comment]
Ah valid points, thanks for the githack link I didn't know that was a thing, that will be helpful fo testing. Ideally I hope to make this change in such a way that it can produce the same behaviour in legitimate situations so that we don't need a migration guide at all, just a fix patch note. 

I think it may be possible to solve the issue with the Clock class but might need some additional fiddling to take into account the special cases you mentioned (tab going to sleep, user disabling/enabling autorotate). I'll investigate some solutions soon.

**Comment by Issung** [comment]
Tried another impl to fix those issues but it's still not super consistent, I would like to use that Timer class but it's been in PR for a while, any idea when it will get merged?

**Comment by Mugen87** [comment]
>  I would like to use that Timer class but it's been in PR for a while, any idea when it will get merged?

I think we should finally merge the PR since I've seen other projects adopting the exact same class or use something similar with a slightly different interface.

It does not make sense to leak the implementation of the `Timer` class into other components like `OrbitControls`. What I mean by that is that `OrbitControls` should not add `visibilitychange` handler but let a time class handle this aspect.

@Issung `autoRotate` requires the call of `OrbitControls.update()` in the animation loop. When using `Timer`, we could call its `update()` method in `OrbitControls.update()` and then just request the time delta in `getAutoRotationAngle()`. The change in `onPointerUp()` should not be necessary anymore. Does this sound right to you^^?

**Comment by yomotsu** [comment]
@Mugen87 

Thanks for your interest.
The reason for it is there are several cases to consider, like:

- Users may want to rotate with a sine curve for the polar axis(like waving)
- Rotate with various controlled speeds (like, slow down in the front and first in the back.
- What if users drag while auto-rotating?

These depend on user requirements, and ended up I thought auto-rotate is not simple and should be handled in user-land.

**Comment by yomotsu** [comment]
By the way, can we just add a delta time argument to `OrbitControl.prototype.update( delta )`?
Then if the argument is undefined, just works as it is now.
So, we don't need an internal `Clock` or `Timer`.

**Comment by Mugen87** [comment]
That is a good point!

@Issung Would you be okay with that solution?

**Comment by Issung** [comment]
Passing the delta time to the update method is a nice non-breaking change that could be made, but it still has the issue of when the tab is put in the background for a few seconds, when coming back to the tab it will have a large visible jump because of the large delta time.

**Comment by Issung** [comment]
It might be a good interim solution while we wait for the Timer class PR to be merged, I could make another PR for that if you like?

**Comment by Mugen87** [comment]
Yes, it would be good to update the PR. If the time delta value is eventually computed with `Timer` the large delta issues should go away.

**Comment by Mugen87** [comment]
THB, I don't think we have to integrate timer if the `update()` approach works fine. Computing a time delta value once and reuse it in the app is actually favorable than computing various deltas in different modules.

**Comment by Issung** [comment]
It seems my update to webgl2_volume_instancing is what is making the tests fail, I'll await your reply on my above question to see what action we should take

**Comment by Mugen87** [comment]
Try to regenerate the E2E screenshot via:
```
npm run make-screenshot webgl2_volume_instancing
```

**Comment by Mugen87** [comment]
I'll merge and clean up.

**Pull request merged**